implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -52,185 +52,226 @@
  * This implements the MP-JWT 1.1 mp.jwt.verify.publickey.location config property resolution logic
  */
 public class KeyLocationResolver implements VerificationKeyResolver {
-    private static final Logger log = Logger.getLogger(KeyLocationResolver.class);
+    private static final Logger LOGGER = Logger.getLogger(KeyLocationResolver.class);
+    private static final String FAILED_TO_READ_LOCATION_MESSAGE = "Failed to read location as any of JWK, JWKS, PEM. ";
 
     // The 'content' and 'httpsJwks' fields are used to keep the key content and mutually exclusive.
     // 'content' represents the key(s) loaded from all resources but the HTTPS URL based JWK set.
-    private String content;
+    private final Supplier<String> contentSupplier;
```

Comment:
Please, just give us back the plain `String` and `HttpsJwk` references, and the original loadContent(), these fancy wrappers make me nervous because I don't what they are to do with the issue this PR is trying to address :-), And those two suppliers add a lot of code too.